src/schema/slack/functions/_scripts/src/write_function_files.ts

```diff
@@ -1,31 +1,47 @@
-import { FunctionsPayload } from "./types.ts";
-import { SlackFunctionModTemplate, SlackFunctionTemplate } from "./template.ts";
+import SlackFunctionTemplate from "./templates/template_function.ts";
+import SlackTestFunctionTemplate from "./templates/test_template.ts";
+import SlackFunctionModTemplate from "./templates/template_mod.ts";
+import { getSlackFunctions, greenText, redText } from "./utils.ts";
+import { FunctionRecord } from "./types.ts";
 
-const yellow = "\x1b[38;5;214m";
-const reset = "\x1b[0m";
-const colorizeText = (text: string) => yellow + text + reset;
+const VALID_FILENAME_REGEX = /^[0-9a-zA-Z_\-]+$/;
 
-const functionsPayload: FunctionsPayload = await Deno.readTextFile(
-  "functions.json",
-).then(JSON.parse);
-
-// Filter out any non slack functions (i.e. has an app_id)
-const slackFunctions = functionsPayload.functions.filter((fn) => !fn.app_id);
+const slackFunctions: FunctionRecord[] = await getSlackFunctions();
 
 // Sorting alphabetically cause only a monster would generate these in a random order
 slackFunctions.sort((a, b) => a.callback_id.localeCompare(b.callback_id));
 
-await slackFunctions.forEach(async (fn) => {
-  console.log(
-    `Generating code for Slack Function: ${colorizeText(fn.callback_id)}`,
-  );
-  const templateString = SlackFunctionTemplate(fn);
-  const filename = `../${fn.callback_id}.ts`;
-
-  await Deno.writeTextFile(filename, templateString);
-});
-
-console.log(`Wrote ${slackFunctions.length} files`);
+await Promise.all(
+  slackFunctions.map(async (functionRecord: FunctionRecord) => {
+    console.log(
+      `Generating code & tests for Slack Function: ${
+        greenText(functionRecord.callback_id)
+      }`,
+    );
+    if (!VALID_FILENAME_REGEX.test(functionRecord.callback_id)) {
+      console.log(
+        `${
+          redText(
+            "Unable to generate files",
+          )
+        }, invalid characters in callback_id: "${functionRecord.callback_id}"`,
+      );
```

Comment:
Oh this is nice! What do you think about
```ts
console.log(
  `${redText("FAILURE:")} Invalid characters in callback_id: ${
    redText(functionRecord.callback_id)
  }`,
);
```

to produce something like
![image](https://user-images.githubusercontent.com/20157895/218615590-f56dea25-91f2-4789-826f-13ea1794101e.png)

Reply:
Ohh yeah thats much better 💯  will push this now

Reply:
Done 🟢